src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -449,39 +449,55 @@ private static FileStream OpenStream(int fd, FileAccess access)
         /// </remarks>
         private static void ParseArgumentsIntoList(string arguments, List<string> results)
         {
-            var currentArgument = new StringBuilder();
-            bool inQuotes = false;
-
             // Iterate through all of the characters in the argument string.
             for (int i = 0; i < arguments.Length; i++)
```

Comment:
This for can be changed to an infinite loop ("for(;;)"). This eliminates a redundant check against argument.Length, and an unnecessary index increment (it also ensures the index never strays outside the bounds 0..argument.Length)